### PR TITLE
Update versions and CHANGELOG

### PR DESCRIPTION
**Pull request by OleksiiZubko**
Updating versions to 1.3.0 and updating CHANGELOG.

Resolves: OLPEDGE-1610

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+**olp-sdk-dataservice-read**
+
+- Added the `VersionedLayerClientParams` interface.
+- Deprecated the `VersionedLayerClient` constructor. Instead, use the constructor that creates the `VersionedLayerClient` instance with the help of the `VersionedLayerClientParams` object.
+- Added the `VersionedLayerClient` constructor that creates the `VersionedLayerClient` instance using the `VersionedLayerClientParams` object and lock version.
+- Deprecated the `DataRequest.getVersion()` method.
+- Deprecated the `DataRequest.withVersion()` method.
+- Changed the `VersionedLayerClient.getData` and `VersionedLayerClient.getPartitions` methods. Now, they ignore versions from `DataRequest` and `PartitionsRequest` and use versions from the `VersionedLayerClient` instance. If you initialize `VersionedLayerClient` without any version, on the first call, the latest version is fetched from the service and set in the instance.
+- Added the `SubscribeRequest` class.
+- Changed `StatisticsRequest`: deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
- Deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+
+**olp-sdk-dataservice-read**
+
+- Added the `VersionedLayerClientParams` interface.
+- Deprecated the `VersionedLayerClient` constructor. Instead, use the constructor that creates the `VersionedLayerClient` instance with the help of the `VersionedLayerClientParams` object.
+- Added the `VersionedLayerClient` constructor that creates the `VersionedLayerClient` instance using the `VersionedLayerClientParams` object and lock version.
+- Deprecated the `DataRequest.getVersion()` method.
+- Deprecated the `DataRequest.withVersion()` method.
+- Changed the `VersionedLayerClient.getData` and `VersionedLayerClient.getPartitions` methods. Now, they ignore versions from `DataRequest` and `PartitionsRequest` and use versions from the `VersionedLayerClient` instance. If you initialize `VersionedLayerClient` without any version, on the first call, the latest version is fetched from the service and set in the instance.
+- Added the `SubscribeRequest` class.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
- Added the `SubscribeRequest` class. This is used by `StreamLayerClient.subscribe()`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+- Changed `StatisticsRequest`: deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
+- Changed `StatisticsRequest.getDataLevel()`. Now, it returns one of the following values: string, number, or undefined.
+- Added `StreamLayerClient`.
+- Added the `StreamLayerClient.subscribe()` method.
+- Added the `StreamLayerClient.poll()` method.
+- Added the `StreamLayerClient.seek()` method.
+- Added the `StreamLayerClient.getData()` method.
+- Added the `StreamLayerClient.unsubscribe()` method.
```
Comment (andescu) [inline CHANGELOG.md]: Merge and describe what is the StreamLayerClient to be used for, e.g. to consume messages from a stream layer in serial or parallel mode, etc.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+- Added the `PollRequest` class.
+- Added the `UnsubscribeRequest` class.
+- Added the `SeekRequest` class.
```
Comment (andescu) [inline CHANGELOG.md]: Mark which method is using this.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,54 @@
+## v1.3.0 (04/03/2020)
+
+**Common**
+
+* Updated dependencies.
+* Improved unit tests coverage.
```
[the remark this answers is not in the document]
Comment (OleksiiZubko) [reply CHANGELOG.md]: removed

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,54 @@
...
+* Fixed bugs.
```
[the remark this answers is not in the document]
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,54 @@
...
+- Updated `node-fetch` to version 2.6.0.
+
+**olp-sdk-authentication**
+
+- Updated `olp-sdk-fetch` to version 1.3.0.
+
+**olp-sdk-dataservice-api**
+
+- Added functions and models that are generated from the Stream API specification.
+- Added unit tests for all public functions.
```
[the remark this answers is not in the document]
Comment (OleksiiZubko) [reply CHANGELOG.md]: done)

File: CHANGELOG.md (hunk 1 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 2 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 4 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 3 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,50 @@
+## v1.3.0 (04/03/2020)
+
+**Common**
+
+- Updated dependencies.
+- Fixed bugs:
+- - Version "0" is incorrectly handled
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
  - Version "0" was incorrectly handled. This is now fixed.
```
Means two spaces and a dash is a sub-list. You should not use a double dash.
Please also place a dot after every sentence.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,50 @@
+## v1.3.0 (04/03/2020)
+
+**Common**
+
+- Updated dependencies.
+- Fixed bugs:
```
Comment (andescu) [inline CHANGELOG.md]: The following bugs are fixed:

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,50 @@
...
+- - Version "0" is incorrectly handled
+- - Additional fields are not passed into request
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
- - Additional fields were not passed correctly into the request. This is now fixed.
```

File: CHANGELOG.md (hunk 8 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: makes sense, updated. Thank you

File: CHANGELOG.md (hunk 9 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 10 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: updated, tnx

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,51 @@
+## v1.3.0 (05/03/2020)
+
+**Common**
+
+- Updated dependencies.
+
+The following bugs are fixed::
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
The following bugs are fixed:
```

Comment (OleksiiZubko) [reply CHANGELOG.md]: omg, updated

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,51 @@
...
+- Added the `SubscribeRequest` class. This is used by `StreamLayerClient.subscribe()`.
+- Deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
+- Changed `StatisticsRequest.getDataLevel()`. Now, it returns one of the following values: string, number, or undefined.
+- Added `StreamLayerClient`. Added reading support for streamed data using the new `StreamLayerClient`. Currently, you can subscribe, unsubscribe, and consume messages from a stream layer in serial or parallel mode. Method `poll` reads the messages from a stream layer and commits successfully consumed messages before handing them over to user. With method `seek` you can set to start reading messages from a stream layer at any given position.
+- Added the `PollRequest` class. This is used by `StreamLayerClient.poll()`.
+- Added the `UnsubscribeRequest` class. This is used by `StreamLayerClient.unsubscribe()`.
+- Added the `SeekRequest` class. This is used by `StreamLayerClient.seek()`.
```
Comment (HalynaDumych) [inline CHANGELOG.md]:
```suggestion
- Added the `SeekRequest` class. It is used by `StreamLayerClient.seek()`.
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,51 @@
...
+- Added the `SubscribeRequest` class. This is used by `StreamLayerClient.subscribe()`.
+- Deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
+- Changed `StatisticsRequest.getDataLevel()`. Now, it returns one of the following values: string, number, or undefined.
+- Added `StreamLayerClient`. Added reading support for streamed data using the new `StreamLayerClient`. Currently, you can subscribe, unsubscribe, and consume messages from a stream layer in serial or parallel mode. Method `poll` reads the messages from a stream layer and commits successfully consumed messages before handing them over to user. With method `seek` you can set to start reading messages from a stream layer at any given position.
+- Added the `PollRequest` class. This is used by `StreamLayerClient.poll()`.
+- Added the `UnsubscribeRequest` class. This is used by `StreamLayerClient.unsubscribe()`.
```
Comment (HalynaDumych) [inline CHANGELOG.md]:
```suggestion
- Added the `UnsubscribeRequest` class. It is used by `StreamLayerClient.unsubscribe()`.
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,51 @@
...
+- Added the `SubscribeRequest` class. This is used by `StreamLayerClient.subscribe()`.
+- Deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
+- Changed `StatisticsRequest.getDataLevel()`. Now, it returns one of the following values: string, number, or undefined.
+- Added `StreamLayerClient`. Added reading support for streamed data using the new `StreamLayerClient`. Currently, you can subscribe, unsubscribe, and consume messages from a stream layer in serial or parallel mode. Method `poll` reads the messages from a stream layer and commits successfully consumed messages before handing them over to user. With method `seek` you can set to start reading messages from a stream layer at any given position.
+- Added the `PollRequest` class. This is used by `StreamLayerClient.poll()`.
```
Comment (HalynaDumych) [inline CHANGELOG.md]:
```suggestion
- Added the `PollRequest` class. It is used by `StreamLayerClient.poll()`.
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,51 @@
...
+- Added the `SubscribeRequest` class. This is used by `StreamLayerClient.subscribe()`.
+- Deprecated the `StatisticsRequest.withDatalevel()` method with a string parameter and added the `StatisticsRequest.withDatalevel()` method with a number parameter.
+- Changed `StatisticsRequest.getDataLevel()`. Now, it returns one of the following values: string, number, or undefined.
+- Added `StreamLayerClient`. Added reading support for streamed data using the new `StreamLayerClient`. Currently, you can subscribe, unsubscribe, and consume messages from a stream layer in serial or parallel mode. Method `poll` reads the messages from a stream layer and commits successfully consumed messages before handing them over to user. With method `seek` you can set to start reading messages from a stream layer at any given position.
```
Comment (HalynaDumych) [inline CHANGELOG.md]:
```suggestion
- Added `StreamLayerClient` and reading support for streamed data using `StreamLayerClient`. Currently, you can subscribe, unsubscribe, and consume messages from a stream layer in a serial or parallel mode. The `poll` method reads messages from a stream layer and commits successfully consumed messages before handing them over to you. With the `seek` method, you can set to start reading messages from a stream layer at any given position.
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,51 @@
+## v1.3.0 (05/03/2020)
+
+**Common**
+
+- Updated dependencies.
+
+The following bugs are fixed:
+- Version "0" was incorrectly handled. This is now fixed.
```
Comment (HalynaDumych) [inline CHANGELOG.md]: It's mentioned inb the intro sentence that this bug is fixed.
```suggestion
- Version "0" was incorrectly handled.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,51 @@
...
+- Additional fields were not passed correctly into the request. This is now fixed.
```
Comment (HalynaDumych) [inline CHANGELOG.md]: It's mentioned inb the intro sentence that this bug is fixed.
```suggestion
- Additional fields were not passed correctly into the request.
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,51 @@
...
+- Added the `UnsubscribeRequest` class. It is used by `StreamLayerClient.unsubscribe()`.
+- Added the `SeekRequest` class. It is used by `StreamLayerClient.seek()`.
+- Deprecated the `VolatileLayerClient`constructor. Instead, use the constructor that creates the `VolatileLayerClient` instance with the help of the `VolatileLayerClientParams` object.
+- Added the `VolatileLayerClient` constructor that creates the `VolatileLayerClient` instance using `VolatileLayerClientClientParams` object.
+- Deprecated the `IndexLayerClient` constructor. Instead, use the constructor that creates the `IndexLayerClientClient` instance with the help of the `IndexLayerClientClientParams` object.
+- Added the `IndexLayerClient` constructor that creates the `IndexLayerClientClient` instance using the `IndexLayerClientClientParams` object.
+- Changed the `LookupAPI.getBaseUrl()` method. Now, you can fetch all URLs in one request instead of fetching each URL separately.
+- Changed the `ConfigClient.getCatalogs()` method. Now, the `request` parameter is optional.
+
+
```
Comment (andescu) [inline CHANGELOG.md]: Remove extra line.

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,51 @@
...
+**Common**
+
+- Updated dependencies.
+
+The following bugs are fixed:
+- Version "0" was incorrectly handled. It is now fixed.
+- Additional fields were not passed correctly into the request. It is now fixed.
+
+
```
Comment (andescu) [inline CHANGELOG.md]: Remove extra line.

File: CHANGELOG.md (hunk 18 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: removed

File: CHANGELOG.md (hunk 19 above)
Comment (OleksiiZubko) [reply CHANGELOG.md]: removed

**Pull request merged**